cdap-test/src/main/java/co/cask/cdap/test/ApplicationManager.java

```diff
@@ -98,6 +98,12 @@
   void stopProgram(ProgramId programId);
 
   /**
+   * Wait for a given programId to have no running run record
+   * @param programId the program id to wait on
+   */
+  void waitForStopped(final ProgramId programId) throws Exception;
```

Comment:
in general if the method just accepts the ProgramId, it should be part of the ProgramManager. it should be easy to move them. file a JIRA as techdebt for this too.